app/services/hyrax/custom_queries/find_related.rb

```diff
@@ -0,0 +1,50 @@
+# Navigate from a resource to the child works in the resource.
+module Hyrax
+  module CustomQueries
+    class FindRelated
+      # Define the queries that can be fulfilled by this custom query.
+      def self.queries
+        [:find_related_for,
+         :find_related_ids_for,
+         :find_inverse_related_for,
+         :find_inverse_related_ids_for]
+      end
+
+      attr_reader :query_service
+
+      def initialize(query_service:)
+        @query_service = query_service
+      end
+
+      # Find related resources, and map to Valkyrie Resources
+      # @param [Valkyrie::Resource]
+      # @return [Array<Valkyrie::Resource>]
+      def find_related_for(resource:, relationship:)
+        find_related_ids_for(resource: resource, relationship: relationship).map { |id| query_service.find_by(id: id) }
```

Comment:
I have a broader question about how these queries refine the built-in Valkyrie queries. Whatever the outcome of that discussion, I think we need to refine this line prior to merge.